packages/world/src/modules/core/CoreModule.sol

```diff
@@ -28,16 +29,27 @@ import { WorldRegistrationSystem } from "./implementations/WorldRegistrationSyst
 
 /**
  * @title Core Module
- * @notice Registers internal World tables, the CoreSystem, and its function selectors.
+ * @notice Registers internal World tables, systems, and their function selectors.
  * @dev This module only supports `installRoot` because it installs root tables, systems and function selectors.
  */
 
 contract CoreModule is Module {
-  /**
-   * @dev Since the CoreSystem only exists once per World and writes to
-   * known tables, we can deploy it once and register it in multiple Worlds.
-   */
-  address internal immutable coreSystem = address(new CoreSystem());
+  address internal immutable accessManagementSystem;
+  address internal immutable balanceTransferSystem;
+  address internal immutable batchCallSystem;
+  address internal immutable coreRegistrationSystem;
+
+  constructor(
+    address _accessManagementSystem,
+    address _balanceTransferSystem,
+    address _batchCallSystem,
+    address _coreRegistrationSystem
```

Comment:
If we made these the respective contract types we'd get improved compile time type checks (right now it would be possible to mess up the argument order without compile time errors). I'm not sure if i'd cause the bytecode size to increase though

Reply:
They are only used as addresses so I'd rather store them as such instead of extra casts within the registration function
But the constructor should indeed receive typed addresses